pinot-broker/src/main/java/com/linkedin/pinot/broker/routing/HelixExternalViewBasedRouting.java

```diff
@@ -555,7 +555,7 @@ public String dumpSnapshot(String tableName)
     JSONArray routingTableSnapshot = new JSONArray();
 
     for (String currentTable : _routingTableBuilderMap.keySet()) {
-      if (tableName == null || currentTable.startsWith(tableName)) {
+      if (tableName == null || tableName.equals(TableNameBuilder.extractRawTableName(currentTable))) {
```

Comment:
why cant we do both? currentTable.startWith || equals. This should preserve the existing behavior

Reply:
Then it'll still return routingTable for `table12` when I only want the routingTable for `table1`, right? I'm not exactly sure whether the existing behavior is a 'feature' or a 'bug' - it is confusing to me when I ask for the routingTable for a table, it's returning information for other unrelated tables. On the UI, I have to look through a whole bunch of unrelated routingTable info in the result to find what I want; when querying from application, I need to do another layer of filtering post getting the result. 

Any anticipated use cases for the current less-strict, prefix-matching logic?

Reply:
Right, we don't really control how users name their tables, and we have tables named `eats_order` and `eats_order_with_count` etc. We are using this endpoint for the Presto-Pinot connector, and this behavior caused some confusion to us.